chips/nrf52/src/uart.rs

```diff
@@ -153,6 +153,8 @@ pub struct Uarte {
     client: Cell<Option<&'static kernel::hil::uart::Client>>,
     buffer: kernel::common::take_cell::TakeCell<'static, [u8]>,
     remaining_bytes: Cell<usize>,
+    rx_buffer: kernel::common::take_cell::TakeCell<'static, [u8]>,
+    rx_remaining_bytes: Cell<usize>,
```

Comment:
Do we need different buffers for tx and rx? Is that handled by the `console capsule`? However, it's no big deal when it comes to memory allocation because `TakeCell` is basically an UnsafeCell that holds a slice so I guess it's safer to keep it.

At least, rename the old `buffer` and `remaining_bytes` to `tx_xx` please :)